src/styles/main.scss

```diff
@@ -0,0 +1,61 @@
+$day-size: 100px;
+$gap-size: 1px;
+$columns: 7;
+$padding-size: 10px;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: $padding-size;
+  gap: $gap-size;
+  max-width: calc(
+    #{$columns} * #{$day-size} + (#{$columns} - 1) * #{$gap-size} + 2 * #{$padding-size}
+  );
+
+  .day {
+    display: flex;
+    justify-content: center;
+    box-sizing: border-box;
+    height: $day-size;
+    width: $day-size;
+    background-color: #eee;
+    border: 1px solid black;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    transition-duration: 500ms;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-of-type(#{$i})::before {
+        content: '#{$i}';
+        display: flex;
+        justify-content: center;
+        align-items: center;
+      }
```

Comment:
The `::before` pseudo-element should be used to add cosmetic content, as stated in the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules). Since the content added here is not purely cosmetic and carries semantic meaning (the day of the month), consider using a different approach, such as an HTML element for the date.